src/lib.rs

```diff
@@ -0,0 +1,389 @@
+}
+
+impl<'a> SuperImport<'a> {
+    /// TODO: Return failure::Error instead if TagError
+    fn read_content_between_tags(&self, chapter_dir: &PathBuf) -> Result<String, TagError> {
+        debug!(
+            r#"Reading content in chapter "{}" for superimport "{:#?}" "#,
+            self.host_chapter.name, self.full_simport_text
+        );
+
+        let path = Path::join(&chapter_dir, &self.file);
+
+        let content = String::from_utf8(::std::fs::read(&path).unwrap()).unwrap();
+
+        // @superimport start foo <--- this line is not captured
+        // ... match all of these
+        // ... lines between the
+        // ... start and end tags <--- this line is not captured
+        // @superimport end foo
+        let start_regex = Regex::new(&format!(
+            r"(?x)         # Insignificant whitespace mode (allows for comments)
+@superimport
+\s+                        # Separating whitespace
+start
+\s+                        # Separating whitespace
+{tag}
+
+.*?                        # Characters between start import tag and end of line
+
+[\n\r]                     # New line right before the start import tag
+
+(?P<content_to_import>     # Everything in between the start and end import lines
+  (.|\n|\r)*
+)
+
+[\n\r]                     # New line right before the end import tag
+
+.*?                        # Characters between start of end import line and end import tag
+
+@superimport
+\s+                        # Separating whitespace
+end
+\s+                        # Separating whitespace
+{tag}
+",
+            tag = regex::escape(self.tag)
+        ))
+        .unwrap();
+
+        let captures = start_regex.captures(&content).unwrap();
+
+        let content_between_tags = captures["content_to_import"].to_string();
+
+        Ok(content_between_tags)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_simports_from_chapter() {
+        let tag_import_chapter = make_tag_import_chapter();
+
+        let simports = SuperImport::find_unescaped_superimports(&tag_import_chapter);
+
+        let expected_simports = vec![SuperImport {
+            host_chapter: &tag_import_chapter,
+            file: "./fixture.css".into(),
+            full_simport_text: "{{#superimport ./fixture.css@cool-css }}",
+            tag: "cool-css",
+            start: 20,
+            end: 60,
+        }];
+
+        assert_eq!(simports, expected_simports);
+    }
+
+    #[test]
+    fn content_between_tags() {
+        let tag_import_chapter = make_tag_import_chapter();
+
+        let simport = &SuperImport::find_unescaped_superimports(&tag_import_chapter)[0];
+
+        let chapter_dir = "book/src/test-cases/tag-import";
+        let chapter_dir = format!("{}/{}", env!("CARGO_MANIFEST_DIR"), chapter_dir);
+
+        let content_between_tags = simport.read_content_between_tags(&chapter_dir.into());
+
+        let expected_content = r#"
+.this-will-be-included {
+  display: block;
+}
+"#;
```

Comment:
🍟 This is a pure preference thing but i'm always worried some junior or uninitiated dev coming along and being like "whoa that looks weird let me indent these lines" and ruin the tests. I'd either add a `/// Intentional lack of indenting` or do this:

```
let expected_content = r#"\n.this-will-be-included {\n  display: block;\n}\n"#;
```

or if valid:
```
let expected_content = r#"\n" +
     ".this-will-be-included {\n" +
         "display: block;\n" +
     "}\n"#;
```

Reply:
I guess tests would fail and said dev would see that - but I'm all for making things more clear!

a0c248d